src/chainstate/burn/distribution.rs

```diff
@@ -38,195 +39,301 @@ use util::log;
+        let mut block_height_at_index = None;
+        for (index, (commits, burns)) in block_commits.iter().zip(user_burns.iter()).enumerate() {
+            let index = index as u64;
+            for commit in commits.iter() {
+                if let Some((first_block_height, first_index)) = block_height_at_index {
+                    assert_eq!(
+                        commit.block_height,
+                        first_block_height + (index - first_index),
+                        "Commits and Burns should be in block height order"
                     );
+                } else {
+                    block_height_at_index = Some((commit.block_height, index));
+                }
+            }
+            for burn in burns.iter() {
+                if let Some((first_block_height, first_index)) = block_height_at_index {
+                    assert_eq!(
+                        burn.block_height,
+                        first_block_height + (index - first_index),
+                        "Commits and Burns should be in block height order"
+                    );
+                } else {
+                    block_height_at_index = Some((burn.block_height, index));
                 }
-                Some(_) => {}
             }
         }
 
-        // consume block candidates and leader keys to produce the list of burn sample points.
-        // If multiple commits try to use the same key, then take the first one (by vtxindex order)
-        // and drop the rest that consume the same key.
-        let mut burn_sample = Vec::with_capacity(all_block_candidates.len());
-        for i in 0..all_block_candidates.len() {
-            let bc = &all_block_candidates[i];
-            match key_index.get(&(bc.key_block_ptr as u64, bc.key_vtxindex as u32)) {
-                None => {
-                    // leader key already consumed; drop this commit
-                    warn!("VRF public key at {},{} already consumed; ignoring block commit {},{} ({})", bc.key_block_ptr, bc.key_vtxindex, bc.block_height, bc.vtxindex, bc.txid);
-                    continue;
-                }
-                Some(i) => {
-                    // next burn sample
-                    burn_sample.push(BurnSamplePoint {
-                        burns: bc.burn_fee as u128, // Initial burn weight is the block commitment's burn
-                        range_start: Uint256::zero(), // To be filled in
-                        range_end: Uint256::zero(), // To be filled in
-                        candidate: bc.clone(),
-                        key: consumed_leader_keys[*i].clone(),
-                        user_burns: vec![],
+        let window_size = block_commits.len() as u8;
+
+        // first, let's link all of the current block commits to the priors
+        let mut commits_with_priors: Vec<_> =
+            // start with the most recent
+            block_commits
+            .remove((window_size - 1) as usize)
```

Comment:
I'm a little confused by this -- are we dropping the current sortition's block commit here?

Reply:
No -- it's pulling it out of the input array to begin building an array with all the linked commits attached.